2024.2.0 build fails on Windows 10 (runs out of memory)

Older versions worked, but with latest one I can't build it anymore. It gets stuck at `testsuite/a11y/names.exe`, uses ALL system memory (I gave 20 cores and 32G of RAM to the VM) and after a while crashes like this:

<details>

```
(tar) Exporting gtk4
Building project gtk4 (4.12.5)
[1/149] Linking target testsuite/gtk/propertylookuplistmodel.exe
FAILED: testsuite/gtk/propertylookuplistmodel.exe testsuite/gtk/propertylookuplistmodel.pdb
"link"  /MACHINE:x64 /OUT:testsuite/gtk/propertylookuplistmodel.exe testsuite/gtk/propertylookuplistmodel.exe.p/propertylookuplistmodel.c.obj "/release" "/nologo" "/DEBUG" "/PDB:testsuite\gtk\propertylookuplistmodel.pdb" "gtk/libgtk.a" "gtk/css/libgtk_css.a" "gdk/libgdk.a" "gsk/libgsk.a" "gsk/libgsk_f16c.a" "C:/gtk-build/gtk/x64/release/lib/gmodule-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/glib-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/intl.lib" "C:/gtk-build/gtk/x64/release/lib/gobject-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/gio-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/pangocairo-1.0.lib" "C:/gtk-build/gtk/x64/release/lib/pango-1.0.lib" "C:/gtk-build/gtk/x64/release/lib/harfbuzz.lib" "C:/gtk-build/gtk/x64/release/lib/cairo.lib" "C:/gtk-build/gtk/x64/release/lib/fribidi.lib" "C:/gtk-build/gtk/x64/release/lib/cairo-gobject.lib" "C:/gtk-build/gtk/x64/release/lib/gdk_pixbuf-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/epoxy.lib" "C:/gtk-build/gtk/x64/release/lib/graphene-1.0.lib" "C:/gtk-build/gtk/x64/release/lib/pangowin32-1.0.lib" "advapi32.lib" "comctl32.lib" "crypt32.lib" "dwmapi.lib" "imm32.lib" "setupapi.lib" "winmm.lib" "intl.lib" "C:/gtk-build/gtk/x64/release/lib/libpng16.lib" "C:/gtk-build/gtk/x64/release/lib/tiff.lib" "C:/gtk-build/gtk/x64/release/lib/jpeg.lib" "hid.lib" "opengl32.lib" "C:/gtk-build/gtk/x64/release/lib/cairo-script-interpreter.lib" "advapi32.lib" "comctl32.lib" "crypt32.lib" "dwmapi.lib" "imm32.lib" "setupapi.lib" "winmm.lib" "intl.lib" "hid.lib" "opengl32.lib" "hid.lib" "opengl32.lib" "hid.lib" "opengl32.lib" "hid.lib" "opengl32.lib" "/SUBSYSTEM:CONSOLE" "kernel32.lib" "user32.lib" "gdi32.lib" "winspool.lib" "shell32.lib" "ole32.lib" "oleaut32.lib" "uuid.lib" "comdlg32.lib" "advapi32.lib"
   Creating library testsuite\gtk\propertylookuplistmodel.lib and object testsuite\gtk\propertylookuplistmodel.exp
[2/149] Linking target testsuite/gtk/colorutils.exe
FAILED: testsuite/gtk/colorutils.exe testsuite/gtk/colorutils.pdb
"link"  /MACHINE:x64 /OUT:testsuite/gtk/colorutils.exe testsuite/gtk/colorutils.exe.p/colorutils.c.obj "/release" "/nologo" "/DEBUG" "/PDB:testsuite\gtk\colorutils.pdb" "gtk/libgtk.a" "gtk/css/libgtk_css.a" "gdk/libgdk.a" "gsk/libgsk.a" "gsk/libgsk_f16c.a" "C:/gtk-build/gtk/x64/release/lib/gmodule-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/glib-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/intl.lib" "C:/gtk-build/gtk/x64/release/lib/gobject-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/gio-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/pangocairo-1.0.lib" "C:/gtk-build/gtk/x64/release/lib/pango-1.0.lib" "C:/gtk-build/gtk/x64/release/lib/harfbuzz.lib" "C:/gtk-build/gtk/x64/release/lib/cairo.lib" "C:/gtk-build/gtk/x64/release/lib/fribidi.lib" "C:/gtk-build/gtk/x64/release/lib/cairo-gobject.lib" "C:/gtk-build/gtk/x64/release/lib/gdk_pixbuf-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/epoxy.lib" "C:/gtk-build/gtk/x64/release/lib/graphene-1.0.lib" "C:/gtk-build/gtk/x64/release/lib/pangowin32-1.0.lib" "advapi32.lib" "comctl32.lib" "crypt32.lib" "dwmapi.lib" "imm32.lib" "setupapi.lib" "winmm.lib" "intl.lib" "C:/gtk-build/gtk/x64/release/lib/libpng16.lib" "C:/gtk-build/gtk/x64/release/lib/tiff.lib" "C:/gtk-build/gtk/x64/release/lib/jpeg.lib" "hid.lib" "opengl32.lib" "C:/gtk-build/gtk/x64/release/lib/cairo-script-interpreter.lib" "advapi32.lib" "comctl32.lib" "crypt32.lib" "dwmapi.lib" "imm32.lib" "setupapi.lib" "winmm.lib" "intl.lib" "hid.lib" "opengl32.lib" "hid.lib" "opengl32.lib" "hid.lib" "opengl32.lib" "hid.lib" "opengl32.lib" "/SUBSYSTEM:CONSOLE" "kernel32.lib" "user32.lib" "gdi32.lib" "winspool.lib" "shell32.lib" "ole32.lib" "oleaut32.lib" "uuid.lib" "comdlg32.lib" "advapi32.lib"
   Creating library testsuite\gtk\colorutils.lib and object testsuite\gtk\colorutils.exp
[3/149] Linking target testsuite/gsk/diff.exe
FAILED: testsuite/gsk/diff.exe testsuite/gsk/diff.pdb
"link"  /MACHINE:x64 /OUT:testsuite/gsk/diff.exe testsuite/gsk/diff.exe.p/diff.c.obj "/release" "/nologo" "/DEBUG" "/PDB:testsuite\gsk\diff.pdb" "gtk/libgtk.a" "gtk/css/libgtk_css.a" "gdk/libgdk.a" "gsk/libgsk.a" "gsk/libgsk_f16c.a" "C:/gtk-build/gtk/x64/release/lib/gmodule-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/glib-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/intl.lib" "C:/gtk-build/gtk/x64/release/lib/gobject-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/gio-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/pangocairo-1.0.lib" "C:/gtk-build/gtk/x64/release/lib/pango-1.0.lib" "C:/gtk-build/gtk/x64/release/lib/harfbuzz.lib" "C:/gtk-build/gtk/x64/release/lib/cairo.lib" "C:/gtk-build/gtk/x64/release/lib/fribidi.lib" "C:/gtk-build/gtk/x64/release/lib/cairo-gobject.lib" "C:/gtk-build/gtk/x64/release/lib/gdk_pixbuf-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/epoxy.lib" "C:/gtk-build/gtk/x64/release/lib/graphene-1.0.lib" "C:/gtk-build/gtk/x64/release/lib/pangowin32-1.0.lib" "advapi32.lib" "comctl32.lib" "crypt32.lib" "dwmapi.lib" "imm32.lib" "setupapi.lib" "winmm.lib" "intl.lib" "C:/gtk-build/gtk/x64/release/lib/libpng16.lib" "C:/gtk-build/gtk/x64/release/lib/tiff.lib" "C:/gtk-build/gtk/x64/release/lib/jpeg.lib" "hid.lib" "opengl32.lib" "C:/gtk-build/gtk/x64/release/lib/cairo-script-interpreter.lib" "advapi32.lib" "comctl32.lib" "crypt32.lib" "dwmapi.lib" "imm32.lib" "setupapi.lib" "winmm.lib" "intl.lib" "hid.lib" "opengl32.lib" "hid.lib" "opengl32.lib" "hid.lib" "opengl32.lib" "hid.lib" "opengl32.lib" "/SUBSYSTEM:CONSOLE" "kernel32.lib" "user32.lib" "gdi32.lib" "winspool.lib" "shell32.lib" "ole32.lib" "oleaut32.lib" "uuid.lib" "comdlg32.lib" "advapi32.lib"
   Creating library testsuite\gsk\diff.lib and object testsuite\gsk\diff.exp
[4/149] Linking target testsuite/gdk/image.exe
FAILED: testsuite/gdk/image.exe testsuite/gdk/image.pdb
"link"  /MACHINE:x64 /OUT:testsuite/gdk/image.exe testsuite/gdk/image.exe.p/image.c.obj "/release" "/nologo" "/DEBUG" "/PDB:testsuite\gdk\image.pdb" "gtk/libgtk.a" "gtk/css/libgtk_css.a" "gdk/libgdk.a" "gsk/libgsk.a" "gsk/libgsk_f16c.a" "C:/gtk-build/gtk/x64/release/lib/gmodule-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/glib-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/intl.lib" "C:/gtk-build/gtk/x64/release/lib/gobject-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/gio-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/pangocairo-1.0.lib" "C:/gtk-build/gtk/x64/release/lib/pango-1.0.lib" "C:/gtk-build/gtk/x64/release/lib/harfbuzz.lib" "C:/gtk-build/gtk/x64/release/lib/cairo.lib" "C:/gtk-build/gtk/x64/release/lib/fribidi.lib" "C:/gtk-build/gtk/x64/release/lib/cairo-gobject.lib" "C:/gtk-build/gtk/x64/release/lib/gdk_pixbuf-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/epoxy.lib" "C:/gtk-build/gtk/x64/release/lib/graphene-1.0.lib" "C:/gtk-build/gtk/x64/release/lib/pangowin32-1.0.lib" "advapi32.lib" "comctl32.lib" "crypt32.lib" "dwmapi.lib" "imm32.lib" "setupapi.lib" "winmm.lib" "intl.lib" "C:/gtk-build/gtk/x64/release/lib/libpng16.lib" "C:/gtk-build/gtk/x64/release/lib/tiff.lib" "C:/gtk-build/gtk/x64/release/lib/jpeg.lib" "hid.lib" "opengl32.lib" "C:/gtk-build/gtk/x64/release/lib/cairo-script-interpreter.lib" "advapi32.lib" "comctl32.lib" "crypt32.lib" "dwmapi.lib" "imm32.lib" "setupapi.lib" "winmm.lib" "intl.lib" "hid.lib" "opengl32.lib" "hid.lib" "opengl32.lib" "hid.lib" "opengl32.lib" "hid.lib" "opengl32.lib" "/SUBSYSTEM:CONSOLE" "kernel32.lib" "user32.lib" "gdi32.lib" "winspool.lib" "shell32.lib" "ole32.lib" "oleaut32.lib" "uuid.lib" "comdlg32.lib" "advapi32.lib"
   Creating library testsuite\gdk\image.lib and object testsuite\gdk\image.exp

libgtk.a(gtktextsegment.c.obj) : fatal error LNK1000: Internal error during IMAGE::Pass2
[5/149] Linking target testsuite/css/transition.exe
FAILED: testsuite/css/transition.exe testsuite/css/transition.pdb
"link"  /MACHINE:x64 /OUT:testsuite/css/transition.exe testsuite/css/transition.exe.p/transition.c.obj "/release" "/nologo" "/DEBUG" "/PDB:testsuite\css\transition.pdb" "gtk/libgtk.a" "gtk/css/libgtk_css.a" "gdk/libgdk.a" "gsk/libgsk.a" "gsk/libgsk_f16c.a" "C:/gtk-build/gtk/x64/release/lib/gmodule-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/glib-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/intl.lib" "C:/gtk-build/gtk/x64/release/lib/gobject-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/gio-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/pangocairo-1.0.lib" "C:/gtk-build/gtk/x64/release/lib/pango-1.0.lib" "C:/gtk-build/gtk/x64/release/lib/harfbuzz.lib" "C:/gtk-build/gtk/x64/release/lib/cairo.lib" "C:/gtk-build/gtk/x64/release/lib/fribidi.lib" "C:/gtk-build/gtk/x64/release/lib/cairo-gobject.lib" "C:/gtk-build/gtk/x64/release/lib/gdk_pixbuf-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/epoxy.lib" "C:/gtk-build/gtk/x64/release/lib/graphene-1.0.lib" "C:/gtk-build/gtk/x64/release/lib/pangowin32-1.0.lib" "advapi32.lib" "comctl32.lib" "crypt32.lib" "dwmapi.lib" "imm32.lib" "setupapi.lib" "winmm.lib" "intl.lib" "C:/gtk-build/gtk/x64/release/lib/libpng16.lib" "C:/gtk-build/gtk/x64/release/lib/tiff.lib" "C:/gtk-build/gtk/x64/release/lib/jpeg.lib" "hid.lib" "opengl32.lib" "C:/gtk-build/gtk/x64/release/lib/cairo-script-interpreter.lib" "advapi32.lib" "comctl32.lib" "crypt32.lib" "dwmapi.lib" "imm32.lib" "setupapi.lib" "winmm.lib" "intl.lib" "hid.lib" "opengl32.lib" "hid.lib" "opengl32.lib" "hid.lib" "opengl32.lib" "hid.lib" "opengl32.lib" "/SUBSYSTEM:CONSOLE" "kernel32.lib" "user32.lib" "gdi32.lib" "winspool.lib" "shell32.lib" "ole32.lib" "oleaut32.lib" "uuid.lib" "comdlg32.lib" "advapi32.lib"
   Creating library testsuite\css\transition.lib and object testsuite\css\transition.exp
[6/149] Linking target testsuite/gtk/textbuffer.exe
FAILED: testsuite/gtk/textbuffer.exe testsuite/gtk/textbuffer.pdb
"link"  /MACHINE:x64 /OUT:testsuite/gtk/textbuffer.exe testsuite/gtk/textbuffer.exe.p/textbuffer.c.obj "/release" "/nologo" "/DEBUG" "/PDB:testsuite\gtk\textbuffer.pdb" "gtk/libgtk.a" "gtk/css/libgtk_css.a" "gdk/libgdk.a" "gsk/libgsk.a" "gsk/libgsk_f16c.a" "C:/gtk-build/gtk/x64/release/lib/gmodule-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/glib-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/intl.lib" "C:/gtk-build/gtk/x64/release/lib/gobject-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/gio-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/pangocairo-1.0.lib" "C:/gtk-build/gtk/x64/release/lib/pango-1.0.lib" "C:/gtk-build/gtk/x64/release/lib/harfbuzz.lib" "C:/gtk-build/gtk/x64/release/lib/cairo.lib" "C:/gtk-build/gtk/x64/release/lib/fribidi.lib" "C:/gtk-build/gtk/x64/release/lib/cairo-gobject.lib" "C:/gtk-build/gtk/x64/release/lib/gdk_pixbuf-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/epoxy.lib" "C:/gtk-build/gtk/x64/release/lib/graphene-1.0.lib" "C:/gtk-build/gtk/x64/release/lib/pangowin32-1.0.lib" "advapi32.lib" "comctl32.lib" "crypt32.lib" "dwmapi.lib" "imm32.lib" "setupapi.lib" "winmm.lib" "intl.lib" "C:/gtk-build/gtk/x64/release/lib/libpng16.lib" "C:/gtk-build/gtk/x64/release/lib/tiff.lib" "C:/gtk-build/gtk/x64/release/lib/jpeg.lib" "hid.lib" "opengl32.lib" "C:/gtk-build/gtk/x64/release/lib/cairo-script-interpreter.lib" "advapi32.lib" "comctl32.lib" "crypt32.lib" "dwmapi.lib" "imm32.lib" "setupapi.lib" "winmm.lib" "intl.lib" "hid.lib" "opengl32.lib" "hid.lib" "opengl32.lib" "hid.lib" "opengl32.lib" "hid.lib" "opengl32.lib" "/SUBSYSTEM:CONSOLE" "kernel32.lib" "user32.lib" "gdi32.lib" "winspool.lib" "shell32.lib" "ole32.lib" "oleaut32.lib" "uuid.lib" "comdlg32.lib" "advapi32.lib"
   Creating library testsuite\gtk\textbuffer.lib and object testsuite\gtk\textbuffer.exp
[7/149] Linking target testsuite/gsk/misc.exe
FAILED: testsuite/gsk/misc.exe testsuite/gsk/misc.pdb
"link"  /MACHINE:x64 /OUT:testsuite/gsk/misc.exe testsuite/gsk/misc.exe.p/misc.c.obj "/release" "/nologo" "/DEBUG" "/PDB:testsuite\gsk\misc.pdb" "gtk/libgtk.a" "gtk/css/libgtk_css.a" "gdk/libgdk.a" "gsk/libgsk.a" "gsk/libgsk_f16c.a" "C:/gtk-build/gtk/x64/release/lib/gmodule-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/glib-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/intl.lib" "C:/gtk-build/gtk/x64/release/lib/gobject-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/gio-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/pangocairo-1.0.lib" "C:/gtk-build/gtk/x64/release/lib/pango-1.0.lib" "C:/gtk-build/gtk/x64/release/lib/harfbuzz.lib" "C:/gtk-build/gtk/x64/release/lib/cairo.lib" "C:/gtk-build/gtk/x64/release/lib/fribidi.lib" "C:/gtk-build/gtk/x64/release/lib/cairo-gobject.lib" "C:/gtk-build/gtk/x64/release/lib/gdk_pixbuf-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/epoxy.lib" "C:/gtk-build/gtk/x64/release/lib/graphene-1.0.lib" "C:/gtk-build/gtk/x64/release/lib/pangowin32-1.0.lib" "advapi32.lib" "comctl32.lib" "crypt32.lib" "dwmapi.lib" "imm32.lib" "setupapi.lib" "winmm.lib" "intl.lib" "C:/gtk-build/gtk/x64/release/lib/libpng16.lib" "C:/gtk-build/gtk/x64/release/lib/tiff.lib" "C:/gtk-build/gtk/x64/release/lib/jpeg.lib" "hid.lib" "opengl32.lib" "C:/gtk-build/gtk/x64/release/lib/cairo-script-interpreter.lib" "advapi32.lib" "comctl32.lib" "crypt32.lib" "dwmapi.lib" "imm32.lib" "setupapi.lib" "winmm.lib" "intl.lib" "hid.lib" "opengl32.lib" "hid.lib" "opengl32.lib" "hid.lib" "opengl32.lib" "hid.lib" "opengl32.lib" "/SUBSYSTEM:CONSOLE" "kernel32.lib" "user32.lib" "gdi32.lib" "winspool.lib" "shell32.lib" "ole32.lib" "oleaut32.lib" "uuid.lib" "comdlg32.lib" "advapi32.lib"
   Creating library testsuite\gsk\misc.lib and object testsuite\gsk\misc.exp

libgtk.a(gtktextutil.c.obj) : fatal error LNK1000: Internal error during IMAGE::Pass2
[8/149] Linking target testsuite/gdk/gltexture.exe
FAILED: testsuite/gdk/gltexture.exe testsuite/gdk/gltexture.pdb
"link"  /MACHINE:x64 /OUT:testsuite/gdk/gltexture.exe testsuite/gdk/gltexture.exe.p/gltexture.c.obj "/release" "/nologo" "/DEBUG" "/PDB:testsuite\gdk\gltexture.pdb" "gtk/libgtk.a" "gtk/css/libgtk_css.a" "gdk/libgdk.a" "gsk/libgsk.a" "gsk/libgsk_f16c.a" "C:/gtk-build/gtk/x64/release/lib/gmodule-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/glib-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/intl.lib" "C:/gtk-build/gtk/x64/release/lib/gobject-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/gio-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/pangocairo-1.0.lib" "C:/gtk-build/gtk/x64/release/lib/pango-1.0.lib" "C:/gtk-build/gtk/x64/release/lib/harfbuzz.lib" "C:/gtk-build/gtk/x64/release/lib/cairo.lib" "C:/gtk-build/gtk/x64/release/lib/fribidi.lib" "C:/gtk-build/gtk/x64/release/lib/cairo-gobject.lib" "C:/gtk-build/gtk/x64/release/lib/gdk_pixbuf-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/epoxy.lib" "C:/gtk-build/gtk/x64/release/lib/graphene-1.0.lib" "C:/gtk-build/gtk/x64/release/lib/pangowin32-1.0.lib" "advapi32.lib" "comctl32.lib" "crypt32.lib" "dwmapi.lib" "imm32.lib" "setupapi.lib" "winmm.lib" "intl.lib" "C:/gtk-build/gtk/x64/release/lib/libpng16.lib" "C:/gtk-build/gtk/x64/release/lib/tiff.lib" "C:/gtk-build/gtk/x64/release/lib/jpeg.lib" "hid.lib" "opengl32.lib" "C:/gtk-build/gtk/x64/release/lib/cairo-script-interpreter.lib" "advapi32.lib" "comctl32.lib" "crypt32.lib" "dwmapi.lib" "imm32.lib" "setupapi.lib" "winmm.lib" "intl.lib" "hid.lib" "opengl32.lib" "hid.lib" "opengl32.lib" "hid.lib" "opengl32.lib" "hid.lib" "opengl32.lib" "/SUBSYSTEM:CONSOLE" "kernel32.lib" "user32.lib" "gdi32.lib" "winspool.lib" "shell32.lib" "ole32.lib" "oleaut32.lib" "uuid.lib" "comdlg32.lib" "advapi32.lib"
   Creating library testsuite\gdk\gltexture.lib and object testsuite\gdk\gltexture.exp
libgdk.a(gdkdrop-win32.c.obj) : warning LNK4099: PDB '' was not found with 'libgdk.a(gdkdrop-win32.c.obj)' or at ''; linking object as if no debug infolibgdk.a(gdkframeclockidle.c.obj) : warning LNK4099: PDB '' was not found with 'libgdk.a(gdkframeclockidle.c.obj)' or at ''; linking object as if no debug info
libgdk.a(gdkinput-dmanipulation.c.obj) : warning LNK4099: PDB '' was not found with 'libgdk.a(gdkinput-dmanipulation.c.obj)' or at ''; linking object as if no debug info
libgdk.a(gdkinput-winpointer.c.obj) : warning LNK4099: PDB '' was not found with 'libgdk.a(gdkinput-winpointer.c.obj)' or at ''; linking object as if no debug info
libgdk.a(gdkdevice-win32.c.obj) : warning LNK4099: PDB '' was not found with 'libgdk.a(gdkdevice-win32.c.obj)' or at ''; linking object as if no debug info
libgdk.a(gdkdevice-virtual.c.obj) : warning LNK4099: PDB '' was not found with 'libgdk.a(gdkdevice-virtual.c.obj)' or at ''; linking object as if no debug info
libgdk.a(gdkdevice-wintab.c.obj) : warning LNK4099: PDB '' was not found with 'libgdk.a(gdkdevice-wintab.c.obj)' or at ''; linking object as if no debug info
libgdk.a(gdkseatdefault.c.obj) : warning LNK4099: PDB '' was not found with 'libgdk.a(gdkseatdefault.c.obj)' or at ''; linking object as if no debug info
libgdk.a(gdkclipdrop-win32.c.obj) : warning LNK4099: PDB '' was not found with 'libgdk.a(gdkclipdrop-win32.c.obj)' or at ''; linking object as if no debug info
libgdk.a(gdkhdataoutputstream-win32.c.obj) : warning LNK4099: PDB '' was not found with 'libgdk.a(gdkhdataoutputstream-win32.c.obj)' or at ''; linking object as if no debug info
libgdk.a(gdkkeys-win32-impl.c.obj) : warning LNK4099: PDB '' was not found with 'libgdk.a(gdkkeys-win32-impl.c.obj)' or at ''; linking object as if no debug info
libgdk.a(gdkglcontext-win32-wgl-private.c.obj) : warning LNK4099: PDB '' was not found with 'libgdk.a(gdkglcontext-win32-wgl-private.c.obj)' or at ''; linking object as if no debug info
libgdk.a(gdkdevice-winpointer.c.obj) : warning LNK4099: PDB '' was not found with 'libgdk.a(gdkdevice-winpointer.c.obj)' or at ''; linking object as if no debug info
libgsk.a(gskglshader.c.obj) : warning LNK4099: PDB '' was not found with 'libgsk.a(gskglshader.c.obj)' or at ''; linking object as if no debug info
libgsk.a(gskrendernode.c.obj) : warning LNK4099: PDB '' was not found with 'libgsk.a(gskrendernode.c.obj)' or at ''; linking object as if no debug info[9/149] Linking target gtk/gtk-4-1.dll
FAILED: gtk/gtk-4-1.dll gtk/gtk-4-1.pdb
"link"  /MACHINE:x64 /OUT:gtk/gtk-4-1.dll  "/release" "/nologo" "/DEBUG" "/PDB:gtk\gtk-4-1.pdb" "/DLL" "/IMPLIB:gtk\gtk-4.lib" "/WHOLEARCHIVE:gtk/libgtk.a" "/WHOLEARCHIVE:gtk/css/libgtk_css.a" "/WHOLEARCHIVE:gdk/libgdk.a" "/WHOLEARCHIVE:gsk/libgsk.a" "gtk/libgtk.a" "gtk/css/libgtk_css.a" "gdk/libgdk.a" "gsk/libgsk.a" "gsk/libgsk_f16c.a" "C:/gtk-build/gtk/x64/release/lib/gmodule-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/glib-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/intl.lib" "C:/gtk-build/gtk/x64/release/lib/gobject-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/gio-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/pangocairo-1.0.lib" "C:/gtk-build/gtk/x64/release/lib/pango-1.0.lib" "C:/gtk-build/gtk/x64/release/lib/harfbuzz.lib" "C:/gtk-build/gtk/x64/release/lib/cairo.lib" "C:/gtk-build/gtk/x64/release/lib/fribidi.lib" "C:/gtk-build/gtk/x64/release/lib/cairo-gobject.lib" "C:/gtk-build/gtk/x64/release/lib/gdk_pixbuf-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/epoxy.lib" "C:/gtk-build/gtk/x64/release/lib/graphene-1.0.lib" "C:/gtk-build/gtk/x64/release/lib/pangowin32-1.0.lib" "advapi32.lib" "comctl32.lib" "crypt32.lib" "dwmapi.lib" "imm32.lib" "setupapi.lib" "winmm.lib" "intl.lib" "C:/gtk-build/gtk/x64/release/lib/libpng16.lib" "C:/gtk-build/gtk/x64/release/lib/tiff.lib" "C:/gtk-build/gtk/x64/release/lib/jpeg.lib" "hid.lib" "opengl32.lib" "C:/gtk-build/gtk/x64/release/lib/cairo-script-interpreter.lib" "advapi32.lib" "comctl32.lib" "crypt32.lib" "dwmapi.lib" "imm32.lib" "setupapi.lib" "winmm.lib" "intl.lib" "hid.lib" "opengl32.lib" "hid.lib" "opengl32.lib" "hid.lib" "opengl32.lib" "hid.lib" "opengl32.lib" "kernel32.lib" "user32.lib" "gdi32.lib" "winspool.lib" "shell32.lib" "ole32.lib" "oleaut32.lib" "uuid.lib" "comdlg32.lib" "advapi32.lib"
   Creating library gtk\gtk-4.lib and object gtk\gtk-4.exp
LINK : fatal error LNK1102: out of memory
[10/149] Linking target testsuite/a11y/names.exe
FAILED: testsuite/a11y/names.exe testsuite/a11y/names.pdb
"link"  /MACHINE:x64 /OUT:testsuite/a11y/names.exe testsuite/a11y/names.exe.p/names.c.obj "/release" "/nologo" "/DEBUG" "/PDB:testsuite\a11y\names.pdb" "gtk/libgtk.a" "gtk/css/libgtk_css.a" "gdk/libgdk.a" "gsk/libgsk.a" "gsk/libgsk_f16c.a" "C:/gtk-build/gtk/x64/release/lib/gmodule-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/glib-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/intl.lib" "C:/gtk-build/gtk/x64/release/lib/gobject-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/gio-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/pangocairo-1.0.lib" "C:/gtk-build/gtk/x64/release/lib/pango-1.0.lib" "C:/gtk-build/gtk/x64/release/lib/harfbuzz.lib" "C:/gtk-build/gtk/x64/release/lib/cairo.lib" "C:/gtk-build/gtk/x64/release/lib/fribidi.lib" "C:/gtk-build/gtk/x64/release/lib/cairo-gobject.lib" "C:/gtk-build/gtk/x64/release/lib/gdk_pixbuf-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/epoxy.lib" "C:/gtk-build/gtk/x64/release/lib/graphene-1.0.lib" "C:/gtk-build/gtk/x64/release/lib/pangowin32-1.0.lib" "advapi32.lib" "comctl32.lib" "crypt32.lib" "dwmapi.lib" "imm32.lib" "setupapi.lib" "winmm.lib" "intl.lib" "C:/gtk-build/gtk/x64/release/lib/libpng16.lib" "C:/gtk-build/gtk/x64/release/lib/tiff.lib" "C:/gtk-build/gtk/x64/release/lib/jpeg.lib" "hid.lib" "opengl32.lib" "C:/gtk-build/gtk/x64/release/lib/cairo-script-interpreter.lib" "advapi32.lib" "comctl32.lib" "crypt32.lib" "dwmapi.lib" "imm32.lib" "setupapi.lib" "winmm.lib" "intl.lib" "hid.lib" "opengl32.lib" "hid.lib" "opengl32.lib" "hid.lib" "opengl32.lib" "hid.lib" "opengl32.lib" "/SUBSYSTEM:CONSOLE" "kernel32.lib" "user32.lib" "gdi32.lib" "winspool.lib" "shell32.lib" "ole32.lib" "oleaut32.lib" "uuid.lib" "comdlg32.lib" "advapi32.lib"
   Creating library testsuite\a11y\names.lib and object testsuite\a11y\names.exp
[11/149] Linking target testsuite/gtk/composetable.exe
FAILED: testsuite/gtk/composetable.exe testsuite/gtk/composetable.pdb
"link"  /MACHINE:x64 /OUT:testsuite/gtk/composetable.exe testsuite/gtk/composetable.exe.p/composetable.c.obj testsuite/gtk/composetable.exe.p/.._testutils.c.obj "/release" "/nologo" "/DEBUG" "/PDB:testsuite\gtk\composetable.pdb" "gtk/libgtk.a" "gtk/css/libgtk_css.a" "gdk/libgdk.a" "gsk/libgsk.a" "gsk/libgsk_f16c.a" "C:/gtk-build/gtk/x64/release/lib/gmodule-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/glib-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/intl.lib" "C:/gtk-build/gtk/x64/release/lib/gobject-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/gio-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/pangocairo-1.0.lib" "C:/gtk-build/gtk/x64/release/lib/pango-1.0.lib" "C:/gtk-build/gtk/x64/release/lib/harfbuzz.lib" "C:/gtk-build/gtk/x64/release/lib/cairo.lib" "C:/gtk-build/gtk/x64/release/lib/fribidi.lib" "C:/gtk-build/gtk/x64/release/lib/cairo-gobject.lib" "C:/gtk-build/gtk/x64/release/lib/gdk_pixbuf-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/epoxy.lib" "C:/gtk-build/gtk/x64/release/lib/graphene-1.0.lib" "C:/gtk-build/gtk/x64/release/lib/pangowin32-1.0.lib" "advapi32.lib" "comctl32.lib" "crypt32.lib" "dwmapi.lib" "imm32.lib" "setupapi.lib" "winmm.lib" "intl.lib" "C:/gtk-build/gtk/x64/release/lib/libpng16.lib" "C:/gtk-build/gtk/x64/release/lib/tiff.lib" "C:/gtk-build/gtk/x64/release/lib/jpeg.lib" "hid.lib" "opengl32.lib" "C:/gtk-build/gtk/x64/release/lib/cairo-script-interpreter.lib" "advapi32.lib" "comctl32.lib" "crypt32.lib" "dwmapi.lib" "imm32.lib" "setupapi.lib" "winmm.lib" "intl.lib" "hid.lib" "opengl32.lib" "hid.lib" "opengl32.lib" "hid.lib" "opengl32.lib" "hid.lib" "opengl32.lib" "/SUBSYSTEM:CONSOLE" "kernel32.lib" "user32.lib" "gdi32.lib" "winspool.lib" "shell32.lib" "ole32.lib" "oleaut32.lib" "uuid.lib" "comdlg32.lib" "advapi32.lib"
   Creating library testsuite\gtk\composetable.lib and object testsuite\gtk\composetable.exp
[12/149] Linking target testsuite/gsk/rounded-rect.exe
   Creating library testsuite\gsk\rounded-rect.lib and object testsuite\gsk\rounded-rect.exp
[13/149] Linking target testsuite/gtk/a11y.exe
FAILED: testsuite/gtk/a11y.exe testsuite/gtk/a11y.pdb
"link"  /MACHINE:x64 /OUT:testsuite/gtk/a11y.exe testsuite/gtk/a11y.exe.p/a11y.c.obj "/release" "/nologo" "/DEBUG" "/PDB:testsuite\gtk\a11y.pdb" "gtk/libgtk.a" "gtk/css/libgtk_css.a" "gdk/libgdk.a" "gsk/libgsk.a" "gsk/libgsk_f16c.a" "C:/gtk-build/gtk/x64/release/lib/gmodule-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/glib-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/intl.lib" "C:/gtk-build/gtk/x64/release/lib/gobject-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/gio-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/pangocairo-1.0.lib" "C:/gtk-build/gtk/x64/release/lib/pango-1.0.lib" "C:/gtk-build/gtk/x64/release/lib/harfbuzz.lib" "C:/gtk-build/gtk/x64/release/lib/cairo.lib" "C:/gtk-build/gtk/x64/release/lib/fribidi.lib" "C:/gtk-build/gtk/x64/release/lib/cairo-gobject.lib" "C:/gtk-build/gtk/x64/release/lib/gdk_pixbuf-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/epoxy.lib" "C:/gtk-build/gtk/x64/release/lib/graphene-1.0.lib" "C:/gtk-build/gtk/x64/release/lib/pangowin32-1.0.lib" "advapi32.lib" "comctl32.lib" "crypt32.lib" "dwmapi.lib" "imm32.lib" "setupapi.lib" "winmm.lib" "intl.lib" "C:/gtk-build/gtk/x64/release/lib/libpng16.lib" "C:/gtk-build/gtk/x64/release/lib/tiff.lib" "C:/gtk-build/gtk/x64/release/lib/jpeg.lib" "hid.lib" "opengl32.lib" "C:/gtk-build/gtk/x64/release/lib/cairo-script-interpreter.lib" "advapi32.lib" "comctl32.lib" "crypt32.lib" "dwmapi.lib" "imm32.lib" "setupapi.lib" "winmm.lib" "intl.lib" "hid.lib" "opengl32.lib" "hid.lib" "opengl32.lib" "hid.lib" "opengl32.lib" "hid.lib" "opengl32.lib" "/SUBSYSTEM:CONSOLE" "kernel32.lib" "user32.lib" "gdi32.lib" "winspool.lib" "shell32.lib" "ole32.lib" "oleaut32.lib" "uuid.lib" "comdlg32.lib" "advapi32.lib"
   Creating library testsuite\gtk\a11y.lib and object testsuite\gtk\a11y.exp
LINK : fatal error LNK1201: error writing to program database ''; check for insufficient disk space, invalid path, or insufficient privilege
[14/149] Linking target tools/gtk4-encode-symbolic-svg.exe
FAILED: tools/gtk4-encode-symbolic-svg.exe tools/gtk4-encode-symbolic-svg.pdb
"link"  /MACHINE:x64 /OUT:tools/gtk4-encode-symbolic-svg.exe tools/gtk4-encode-symbolic-svg.exe.p/encodesymbolic.c.obj "/release" "/nologo" "/DEBUG" "/PDB:tools\gtk4-encode-symbolic-svg.pdb" "gtk/libgtk.a" "gtk/css/libgtk_css.a" "gdk/libgdk.a" "gsk/libgsk.a" "gsk/libgsk_f16c.a" "C:/gtk-build/gtk/x64/release/lib/gmodule-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/glib-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/intl.lib" "C:/gtk-build/gtk/x64/release/lib/gobject-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/gio-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/pangocairo-1.0.lib" "C:/gtk-build/gtk/x64/release/lib/pango-1.0.lib" "C:/gtk-build/gtk/x64/release/lib/harfbuzz.lib" "C:/gtk-build/gtk/x64/release/lib/cairo.lib" "C:/gtk-build/gtk/x64/release/lib/fribidi.lib" "C:/gtk-build/gtk/x64/release/lib/cairo-gobject.lib" "C:/gtk-build/gtk/x64/release/lib/gdk_pixbuf-2.0.lib" "C:/gtk-build/gtk/x64/release/lib/epoxy.lib" "C:/gtk-build/gtk/x64/release/lib/graphene-1.0.lib" "C:/gtk-build/gtk/x64/release/lib/pangowin32-1.0.lib" "advapi32.lib" "comctl32.lib" "crypt32.lib" "dwmapi.lib" "imm32.lib" "setupapi.lib" "winmm.lib" "intl.lib" "C:/gtk-build/gtk/x64/release/lib/libpng16.lib" "C:/gtk-build/gtk/x64/release/lib/tiff.lib" "C:/gtk-build/gtk/x64/release/lib/jpeg.lib" "hid.lib" "opengl32.lib" "C:/gtk-build/gtk/x64/release/lib/cairo-script-interpreter.lib" "advapi32.lib" "comctl32.lib" "crypt32.lib" "dwmapi.lib" "imm32.lib" "setupapi.lib" "winmm.lib" "intl.lib" "hid.lib" "opengl32.lib" "hid.lib" "opengl32.lib" "hid.lib" "opengl32.lib" "hid.lib" "opengl32.lib" "/SUBSYSTEM:CONSOLE" "kernel32.lib" "user32.lib" "gdi32.lib" "winspool.lib" "shell32.lib" "ole32.lib" "oleaut32.lib" "uuid.lib" "comdlg32.lib" "advapi32.lib"
   Creating library tools\gtk4-encode-symbolic-svg.lib and object tools\gtk4-encode-symbolic-svg.exp
LINK : fatal error LNK1102: out of memory
[15/149] Linking target testsuite/gtk/constraint-solver.exe
   Creating library testsuite\gtk\constraint-solver.lib and object testsuite\gtk\constraint-solver.exp
[16/149] Linking target testsuite/gdk/texture.exe
   Creating library testsuite\gdk\texture.lib and object testsuite\gdk\texture.exp
[17/149] Linking target testsuite/gtk/rbtree.exe
   Creating library testsuite\gtk\rbtree.lib and object testsuite\gtk\rbtree.exp
[18/149] Linking target testsuite/gtk/imcontext.exe
```

</details>

I also don't understand why is it bulding `testsuite` in the first place.

Looks similar https://github.com/wingtk/gvsbuild/issues/1232, where logs seems to be incomplete.

Hi @danyeaw, I am building using `gvsbuild build gtk4` gvsbuild version is v2024.5.0. I aslo tried with `gvsbuild build --ninja-opts '-j 2'  gtk4` , this time it didn't ran out memory but still failed at `FAILED: testsuite/gdk/image.exe testsuite/gdk/image.pdb`. 
Attaching error log below 

> `Building project gtk4 (4.14.4)
> [2/112] Linking target testsuite/gdk/image.exe
> FAILED: testsuite/gdk/image.exe testsuite/gdk/image.pdb
> "link"  /MACHINE:x64 /OUT:testsuite/gdk/image.exe testsuite/gdk/image.exe.p/image.c.obj "/release" "/nologo" "/DEBUG" "/PDB:testsuite\gdk\image.pdb" "gtk/libgtk.a" "gtk/css/libgtk_css.a" "gdk/libgdk.a" "gsk/libgsk.a" "gsk/libgsk_f16c.a" "C:/gtk-build/gtk/x64/release/bin/../lib/gmodule-2.0.lib" "C:/gtk-build/gtk/x64/release/bin/../lib/glib-2.0.lib" "C:/gtk-build/gtk/x64/release/bin/../lib/intl.lib" "C:/gtk-build/gtk/x64/release/bin/../lib/gobject-2.0.lib" "C:/gtk-build/gtk/x64/release/bin/../lib/gio-2.0.lib" "C:/gtk-build/gtk/x64/release/bin/../lib/pangocairo-1.0.lib" "C:/gtk-build/gtk/x64/release/bin/../lib/pango-1.0.lib" "C:/gtk-build/gtk/x64/release/bin/../lib/cairo.lib" "C:/gtk-build/gtk/x64/release/bin/../lib/harfbuzz.lib" "C:/gtk-build/gtk/x64/release/bin/../lib/fribidi.lib" "C:/gtk-build/gtk/x64/release/bin/../lib/cairo-gobject.lib" "C:/gtk-build/gtk/x64/release/bin/../lib/gdk_pixbuf-2.0.lib" "C:/gtk-build/gtk/x64/release/bin/../lib/epoxy.lib" "C:/gtk-build/gtk/x64/release/bin/../lib/graphene-1.0.lib" "C:/gtk-build/gtk/x64/release/bin/../lib/pangowin32-1.0.lib" "advapi32.lib" "comctl32.lib" "crypt32.lib" "dwmapi.lib" "imm32.lib" "setupapi.lib" "winmm.lib" "intl.lib" "C:/gtk-build/gtk/x64/release/bin/../lib/libpng16.lib" "C:/gtk-build/gtk/x64/release/bin/../lib/tiff.lib" "C:/gtk-build/gtk/x64/release/bin/../lib/jpeg.lib" "hid.lib" "opengl32.lib" "C:/gtk-build/gtk/x64/release/bin/../lib/cairo-script-interpreter.lib" "advapi32.lib" "comctl32.lib" "crypt32.lib" "dwmapi.lib" "imm32.lib" "setupapi.lib" "winmm.lib" "intl.lib" "hid.lib" "opengl32.lib" "hid.lib" "opengl32.lib" "hid.lib" "opengl32.lib" "hid.lib" "opengl32.lib" "/SUBSYSTEM:CONSOLE" "kernel32.lib" "user32.lib" "gdi32.lib" "winspool.lib" "shell32.lib" "ole32.lib" "oleaut32.lib" "uuid.lib" "comdlg32.lib" "advapi32.lib"
> LINK : fatal error LNK1285: corrupt PDB file 'C:\gtk-build\build\x64\release\gtk4\_gvsbuild-meson\testsuite\gdk\image.pdb'; delete and rebuild
> [3/112] Linking target testsuite/gdk/texture.exe
>    Creating library testsuite\gdk\texture.lib and object testsuite\gdk\texture.exp
> ninja: build stopped: subcommand failed.
> Traceback (most recent call last):
>   File "C:\Users\tanis\pipx\venvs\gvsbuild\Lib\site-packages\gvsbuild\utils\builder.py", line 527, in build
>     if self.__build_one(p):
>        ^^^^^^^^^^^^^^^^^^^
>   File "C:\Users\tanis\pipx\venvs\gvsbuild\Lib\site-packages\gvsbuild\utils\builder.py", line 663, in __build_one
>     skip_deps = proj.build()
>                 ^^^^^^^^^^^^
>   File "C:\Users\tanis\pipx\venvs\gvsbuild\Lib\site-packages\gvsbuild\projects\gtk.py", line 156, in build
>     Meson.build(self)
>   File "C:\Users\tanis\pipx\venvs\gvsbuild\Lib\site-packages\gvsbuild\utils\base_builders.py", line 56, in build
>     self.builder.exec_ninja(params="install", working_dir=ninja_build)
>   File "C:\Users\tanis\pipx\venvs\gvsbuild\Lib\site-packages\gvsbuild\utils\builder.py", line 985, in exec_ninja
>     self.__execute(
>   File "C:\Users\tanis\pipx\venvs\gvsbuild\Lib\site-packages\gvsbuild\utils\builder.py", line 1041, in __execute
>     subprocess.check_call(args, cwd=working_dir, env=env, shell=True)
>   File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.12_3.12.1008.0_x64__qbz5n2kfra8p0\Lib\subprocess.py", line 413, in check_call
>     raise CalledProcessError(retcode, cmd)
> subprocess.CalledProcessError: Command 'ninja -j 2 install' returned non-zero exit status 1.
> Error: gtk4 build failed`

--- a/README.md
+++ b/README.md
@@ -290,6 +290,8 @@ to build GTK with `gvsbuild build gtk4 adwaita-icon-theme` which will include li
 rebuilding it with `--from-scratch`
 - If the download of a tarball fails a partial file will not pass the hash check,
 delete the file and try again.
+- If you get an out of memory error, reduce the number of processor cores building at once
+by adding the `--ninja-opts -j2` option, where 2 is the number of cores.
 
 ## OpenSSL
 

--- a/gvsbuild/projects/gtk.py
+++ b/gvsbuild/projects/gtk.py
@@ -147,6 +147,7 @@ class Gtk4(Tarball, Meson):
 
         self.add_param(f"-Dintrospection={enable_gi}")
         self.add_param("-Dbuild-tests=false")
+        self.add_param("-Dbuild-testsuite=false")
         self.add_param("-Dbuild-demos=false")
         self.add_param("-Dbuild-examples=false")
         self.add_param("-Dmedia-gstreamer=disabled")

